homeassistant/components/influxdb/config_flow.py

```diff
@@ -0,0 +1,287 @@
+from .const import (
+    API_VERSION_2,
+    CONF_API_VERSION,
+    CONF_BUCKET,
+    CONF_DB_NAME,
+    CONF_ORG,
+    CONF_SSL_CA_CERT,
+    DEFAULT_API_VERSION,
+    DEFAULT_HOST,
+    DEFAULT_PORT,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+INFLUXDB_V1_SCHEMA = vol.Schema(
+    {
+        vol.Required(
+            CONF_URL, default=f"http://{DEFAULT_HOST}:{DEFAULT_PORT}"
+        ): TextSelector(
+            TextSelectorConfig(
+                type=TextSelectorType.URL,
+                autocomplete="url",
+            ),
+        ),
+        vol.Required(CONF_VERIFY_SSL, default=False): bool,
+        vol.Required(CONF_DB_NAME): TextSelector(
+            TextSelectorConfig(
+                type=TextSelectorType.TEXT,
+            ),
+        ),
+        vol.Optional(CONF_USERNAME): TextSelector(
+            TextSelectorConfig(
+                type=TextSelectorType.TEXT,
+                autocomplete="username",
+            ),
+        ),
+        vol.Optional(CONF_PASSWORD): TextSelector(
+            TextSelectorConfig(
+                type=TextSelectorType.PASSWORD,
+                autocomplete="current-password",
+            ),
+        ),
+        vol.Optional(CONF_SSL_CA_CERT): FileSelector(
+            FileSelectorConfig(accept=".pem,.crt,.cer,.der")
+        ),
+    }
+)
+
+INFLUXDB_V2_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_URL, default="https://"): TextSelector(
+            TextSelectorConfig(
+                type=TextSelectorType.URL,
+                autocomplete="url",
+            ),
+        ),
+        vol.Required(CONF_VERIFY_SSL, default=False): bool,
+        vol.Required(CONF_ORG): TextSelector(
+            TextSelectorConfig(
+                type=TextSelectorType.TEXT,
+            ),
+        ),
+        vol.Required(CONF_BUCKET): TextSelector(
+            TextSelectorConfig(
+                type=TextSelectorType.TEXT,
+            ),
+        ),
+        vol.Required(CONF_TOKEN): TextSelector(
+            TextSelectorConfig(
+                type=TextSelectorType.TEXT,
+            ),
+        ),
+        vol.Optional(CONF_SSL_CA_CERT): FileSelector(
+            FileSelectorConfig(accept=".pem,.crt,.cer,.der")
+        ),
+    }
+)
+
+
+async def _validate_influxdb_connection(
+    hass: HomeAssistant, data: dict[str, str]
+) -> dict[str, str]:
+    """Validate connection to influxdb."""
+    errors = {}
+
+    try:
+        influx = await hass.async_add_executor_job(get_influx_connection, data, True)
+        influx.close()
+    except ConnectionError as ex:
+        _LOGGER.error(ex)
+        if "SSLError" in ex.args[0]:
+            errors = {"base": "ssl_error"}
+        elif "database not found" in ex.args[0]:
+            errors = {"base": "invalid_database"}
+        elif "authorization failed" in ex.args[0]:
+            errors = {"base": "invalid_auth"}
+        elif "token" in ex.args[0]:
+            errors = {"base": "invalid_auth_v2"}
```

Comment:
`Invalid organization, bucket or token`

Is that related to auth? Can we pick a better name than invalid_auth_v2?

Reply:
It is somehow related to auth, your token could not have the correct access rights. However, you could also have mistyped the bucket name or organization. Maybe `invalid_config`?